jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/ConstructorRule.java

```diff
@@ -200,38 +198,50 @@ private LinkedHashSet<String> getConstructorProperties(JsonNode node, boolean on
       Map.Entry<String, JsonNode> property = properties.next();
 
       JsonNode propertyObj = property.getValue();
+      final String javadoc = getJavadocForProperty(propertyObj);
       if (onlyRequired) {
         // draft3 style
-        if (propertyObj.has("required") && propertyObj.get("required")
-            .asBoolean()) {
-          rtn.add(nameHelper.getPropertyName(property.getKey(), property.getValue()));
+        if (propertyObj.has("required") && propertyObj.get("required").asBoolean()) {
+            rtn.put(nameHelper.getPropertyName(property.getKey(), property.getValue()), javadoc);
         }
 
         // draft4 style
         if (draft4RequiredProperties.contains(property.getKey())) {
-          rtn.add(nameHelper.getPropertyName(property.getKey(), property.getValue()));
+          rtn.put(nameHelper.getPropertyName(property.getKey(), property.getValue()), javadoc);
         }
       } else {
-        rtn.add(nameHelper.getPropertyName(property.getKey(), property.getValue()));
+        rtn.put(nameHelper.getPropertyName(property.getKey(), property.getValue()), javadoc);
       }
     }
     return rtn;
   }
 
+    private String getJavadocForProperty(JsonNode propertyObj) {
+        final boolean hasDescription = propertyObj.has("description") && !propertyObj.get("description").isNull();
+        final String title;
+        if (propertyObj.has("title") && !propertyObj.get("title").isNull()) {
+            title = StringUtils.appendIfMissing(propertyObj.get("title").asText(), ".") + " ";
+        } else {
+            title = "";
+        }
+        final String description = hasDescription ? StringUtils.appendIfMissing(propertyObj.get("description").asText(), ".") : "";
+        return title + description;
```

Comment:
Sorry to be pedantic, but if we have title only and no description, then we should not have a space after the `.`.

Reply:
Replaced with `StringJoiner`

Reply:
I didn't know about StringJoiner, nice!